### PR TITLE
add: namespace and including

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**

**Changes:**

### CODE REVIEW

File: guides/coding_style/c++/style.md [hunk 1]
```diff
@@ -40,6 +40,9 @@ Including Header Files
 ===
 When including header files, only include files that are _portable_ across all compilers (Unless where applicable). Do not include `<bits/stdc++.h>`.
 
+Please pay attention to that if the header is _not in use_, then do not include it (if only implementation use it, then include the file in .cpp, not .hpp).
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: Reword: Pay attention to whether the header is \_not in use\_, and do not include it if so. If only the implementation uses  it, then include the file in .cpp, not .hpp.

File: guides/coding_style/c++/style.md [hunk 2]
```diff
@@ -70,6 +73,14 @@ public:
 */
 ```
 
+Namespace
+===
+When using namespace, please pay attention to `name collision`(also `name conflict`):
+
+- In header files, do NOT use `using` directive or declaration.
+
+- In implementation files, please also minimize the visibility.
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: Unclear what you mean here.

File: guides/coding_style/c++/style.md [hunk 3]
```diff
@@ -70,6 +73,14 @@ public:
 */
 ```
 
+Namespace
+===
+When using namespace, please pay attention to `name collision`(also `name conflict`):
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: Remove the please, and reword to name collision **_s_**. Also remove the code tags are name collision/name conflict.

File: guides/coding_style/c++/style.md [hunk 4]
```diff
@@ -70,6 +73,14 @@ public:
...
+
+- In header files, do NOT use `using` directive or declaration.
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: Give an example (using namespace std).

File: guides/coding_style/c++/style.md [hunk 5]
```diff
@@ -70,6 +96,37 @@ public:
 */
 ```
 
+Namespaces
+===
+When using namespace, pay attention to `name collisions/name conflicts`):
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: No code tags, and missing opening parentheses.

File: guides/coding_style/c++/style.md [hunk 6]
```diff
@@ -40,6 +40,32 @@ Including Header Files
 ===
 When including header files, only include files that are _portable_ across all compilers (Unless where applicable). Do not include `<bits/stdc++.h>`.
 
+Pay attention to whether the header is _not in use_, and do not include it if so. If only the implementation uses it, then include the file in .cpp, not .hpp.
+
+**Valid**
+```C++
+// sample1.hpp
+#include <vector>       // ok, use the vector be return type
```
Comment (arnavb) [inline guides/coding_style/c++/style.md]: Fix comment grammar.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**